cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java

```diff
@@ -0,0 +1,357 @@
+            ctx.status(HttpServletResponse.SC_INTERNAL_SERVER_ERROR).json(re);
+        }
+
+    }
+
+    @Nullable
+    private static ZonedDateTime queryParamAsZdt(Context ctx, String param, String timezone) {
+        ZonedDateTime beginZdt = null;
+        String begin = ctx.queryParam(param);
+        if (begin != null) {
+            beginZdt = DateUtils.parseUserDate(begin, timezone);
+        }
+        return beginZdt;
+    }
+
+    @Nullable
+    private static ZonedDateTime queryParamAsZdt(Context ctx, String param) {
+        return queryParamAsZdt(ctx, param, ctx.queryParamAsClass(TIMEZONE, String.class).getOrDefault("UTC"));
+    }
+
+    @OpenApi(ignore = true)
+    @Override
+    public void getOne(@NotNull Context ctx, @NotNull String templateId) {
+        throw new UnsupportedOperationException(NOT_SUPPORTED_YET);
+    }
+
+    @OpenApi(
+        description = "Create new TextTimeSeries",
+        requestBody = @OpenApiRequestBody(
+            content = {
+                @OpenApiContent(from = TextTimeSeries.class, type = Formats.JSONV2)
+            },
+            required = true),
+        queryParams = {
+            @OpenApiParam(name = MAX_VERSION, type = Boolean.class, description = "Whether to use the maximum version date if p_version_date is not specified."),
+            @OpenApiParam(name = REPLACE_ALL, type = Boolean.class)
+                        },
+        method = HttpMethod.POST,
+        tags = {TAG}
+    )
+    @Override
+    public void create(@NotNull Context ctx) {
+        try (Timer.Context ignored = markAndTime(CREATE)) {
+            DSLContext dsl = getDslContext(ctx);
+
+            String reqContentType = ctx.req.getContentType();
+            String formatHeader = reqContentType != null ? reqContentType : Formats.JSONV2;
+            String body = ctx.body();
+            TextTimeSeries tts = deserialize(body, formatHeader);
+            TimeSeriesTextDao dao = getDao(dsl);
+
+            boolean maxVersion = ctx.queryParamAsClass(MAX_VERSION, Boolean.class).getOrDefault(false);
+            boolean replaceAll = ctx.queryParamAsClass(REPLACE_ALL, Boolean.class).getOrDefault(false);
+            dao.create(tts, maxVersion, replaceAll);
+            ctx.status(HttpServletResponse.SC_CREATED);
+        } catch (JsonProcessingException ex) {
+            CdaError re = new CdaError("Failed to process create request");
+            logger.log(Level.SEVERE, re.toString(), ex);
+            ctx.status(HttpServletResponse.SC_INTERNAL_SERVER_ERROR).json(re);
+        }
+    }
+
+    @OpenApi(
+        description = "Updates a text timeseries",
+        pathParams = {
+            @OpenApiParam(name = TS_ID, description = "The id of the text timeseries to be updated"),
+        },
+            queryParams = {
+                    @OpenApiParam(name = MAX_VERSION, type = Boolean.class, description = "Whether to use the maximum version date if p_version_date is not specified."),
```

Comment:
If we aren't using a p_version_date yet, we should leave off the `MAX_VERSION` (I'm not even sure what max_version means in the case of storage)

Reply:
I think it means the individual rows.  it the rows don't include a version-date what happens?

Reply:
I don't think anything got ignored but I'll remove the parameter.